server/src/test/java/org/opensearch/search/SearchServiceTests.java

```diff
@@ -823,6 +824,118 @@ public Scroll scroll() {
         }
     }
 
+    public void testQuerySearchResultTookTime() throws Exception {
```

Comment:
I think the right place to add this test should be in QueryPhaseTests. U can create a QueryPhaseObject as that tests has some logic to initialize dependent classes.

See if you can add a deliberate delay X, using QueryPhaseSearcher , you could verify tookTime>=X